backend/prisma/schema.prisma

```diff
@@ -12,22 +12,25 @@ datasource db {
 }
 
 model User {
-  id           String        @id @unique @default(uuid())
-  email        String        @unique
-  password     String
-  firstName    String?
-  lastName     String?
-  role         Role          @default(USER)
-  createdAt    DateTime      @default(now())
-  updatedAt    DateTime      @updatedAt
-  writing      Writing[]
-  referredBy   String?
-  referralCode String        @unique
-  balance      Int           @default(0)
-  transactions Transaction[]
+  id                  String            @id @unique @default(uuid())
+  email               String            @unique
+  password            String
+  firstName           String?
+  lastName            String?
+  phoneNumber         String?
+  birthDate           DateTime?
+  createdAt           DateTime          @default(now())
+  updatedAt           DateTime          @updatedAt
+  referredBy          String?
+  referralCode        String            @unique
+  balance             Int               @default(0)
+  role                Role              @default(USER)
+  writing             Writing[]
+  transactions        BankTransaction[]
+  InternalTransaction AppTransaction[]
 }
 
-model Transaction {
+model BankTransaction {
```

Comment:
Just create a single Transaction table, then add a type to it to configure if it is a local or remote transaction